### PR TITLE
Sdl winport merge to develop with tests (part5)

### PR DESCRIPTION
**Pull request by OHerasym**
Fix policy tests

### CODE REVIEW

**Comment by OHerasym** [comment]
@anosach-luxoft , @dev-gh, @VVeremjova , @AByzhynar, @Kozoriz, @LuxoftAKutsan, @dtrunov, Please review

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 1]
```diff
@@ -104,7 +106,7 @@ class PolicyManagerImplTest : public ::testing::Test {
   NiceMock<MockPolicyListener> listener;
 
   void SetUp() OVERRIDE {
-    manager = new PolicyManagerImpl();
+    manager = new PolicyManagerImpl("testing", 1, 10000);
```
Comment (dev-gh) [inline src/components/policy/test/policy_manager_impl_test.cc]: @OHerasym 
Magic numbers

File: src/components/policy/test/sql_pt_representation_test.cc [hunk 2]
```diff
@@ -73,6 +73,7 @@ namespace policy {
 class SQLPTRepresentationTest : public SQLPTRepresentation,
                                 public ::testing::Test {
  protected:
+  SQLPTRepresentationTest() : SQLPTRepresentation(".", 5, 7000) {}
```
Comment (dev-gh) [inline src/components/policy/test/sql_pt_representation_test.cc]: @OHerasym 
Magic numbers

File: src/components/policy/test/sql_pt_representation_test.cc [hunk 3]
```diff
@@ -200,35 +201,37 @@ class SQLPTRepresentationTest : public SQLPTRepresentation,
...
     module_config["exchange_after_x_days"] = Json::Value(5);
     module_config["timeout_after_x_seconds"] = Json::Value(500);
-    module_config["seconds_between_retries"] = Json::Value(Json::arrayValue);
-    Json::Value& seconds_between_retries =
+    module_config["seconds_between_retries"] =
+        utils::json::ValueType::ARRAY_VALUE;
+    utils::json::JsonValueRef seconds_between_retries =
         module_config["seconds_between_retries"];
-    seconds_between_retries[0] = Json::Value(10);
+    seconds_between_retries[uint32_t(0)] = Json::Value(10);
```
Comment (dev-gh) [inline src/components/policy/test/sql_pt_representation_test.cc]: @OHerasym 
What is the reason?

**Comment by dev-gh** [comment]
Reviewed

File: src/components/policy/test/sql_pt_representation_test.cc (hunk 3 above)
Comment (OHerasym) [inline src/components/policy/test/sql_pt_representation_test.cc]: @dev-gh , QT don't like just [0]

File: src/components/policy/test/policy_manager_impl_test.cc (hunk 1 above)
Comment (OHerasym) [inline src/components/policy/test/policy_manager_impl_test.cc]: @dev-gh , 3aceb8d , Done

File: src/components/policy/test/sql_pt_representation_test.cc (hunk 2 above)
Comment (OHerasym) [inline src/components/policy/test/sql_pt_representation_test.cc]: @dev-gh , 3aceb8d , Done

**Comment by OHerasym** [comment]
@anosach-luxoft , @dev-gh, @VVeremjova , @AByzhynar, @Kozoriz, @LuxoftAKutsan, @dtrunov, Please review changes

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 4]
```diff
@@ -150,12 +156,11 @@ class PolicyManagerImplTest2 : public ::testing::Test {
...
 
   void SetUp() OVERRIDE {
+    DefaultValue<const std::string&>::Set(kAppStorageFolder);
     file_system::CreateDirectory("storage1");
     file_system::DeleteFile("policy.sqlite");
 
-    manager = new PolicyManagerImpl();
-    ON_CALL(policy_settings_, app_storage_folder())
-        .WillByDefault(ReturnRef(kAppStorageFolder));
+    manager = new PolicyManagerImpl("", 1, 10000);
```
Comment (VVeremjova) [inline src/components/policy/test/policy_manager_impl_test.cc]: @OHerasym Magic numbers

File: src/components/policy/test/sql_pt_representation_test.cc [hunk 5]
```diff
@@ -81,10 +90,11 @@ class SQLPTRepresentationTest : public SQLPTRepresentation,
       policy_settings_;
 
   static void SetUpTestCase() {
-    const std::string kAppStorageFolder = "storage1";
-    file_system::RemoveDirectory(kAppStorageFolder);
+    const std::string kAppStorageFolder = ".";
+    file_system::RemoveDirectory(kAppStorageFolder, false);
```
Comment (VVeremjova) [inline src/components/policy/test/sql_pt_representation_test.cc]: @OHerasym Why do you try to remove current directory?

**Comment by AByzhynar** [comment]
Reviewed

**Pull request closed without merging**